src/beta_inc.jl

```diff
@@ -13,6 +13,7 @@ loggammadiv(a::Number, b::Number) = _loggammadiv(float(a), float(b))
 
 _loggammadiv(a::T, b::T) where {T<:Base.IEEEFloat} = T(_loggammadiv(Float64(a), Float64(b))) # handle Float16, Float32
 function _loggammadiv(a::Float64, b::Float64)
+    @assert b >= 8
```

Comment:
I do think it's only for debugging here. These function are only supposed to be used as part of other functions and it will actually be useful that they can be removed by optimization passes.